bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/internal/ConfigMapperTest.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2014,2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.eclipse.smarthome.config.core.internal;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.core.IsCollectionContaining.hasItems;
+import static org.junit.Assert.assertThat;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+
+/**
+ *
+ * @author Henning Treu - initial contribution
+ *
+ */
+@SuppressWarnings("null")
+public class ConfigMapperTest {
+
+    @Test
+    public void testSimpleMapping() {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("string", "MyString");
+        properties.put("integer", Integer.valueOf(10));
+        properties.put("bool", Boolean.TRUE);
+        properties.put("list", Arrays.asList("one", "two", "three"));
+
+        SimpleTestConfiguration configuration = ConfigMapper.as(properties, SimpleTestConfiguration.class);
+
+        assertThat(configuration.string, is("MyString"));
+        assertThat(configuration.integer, is(Integer.valueOf(10)));
+        assertThat(configuration.bool, is(true));
+        assertThat(configuration.list, hasItems("one", "two", "three"));
+    }
+
+    @Test
+    public void testSingleValueListMapping() {
+        Map<String, Object> properties = new HashMap<>();
+        properties.put("list", "one");
+
+        SimpleTestConfiguration configuration = ConfigMapper.as(properties, SimpleTestConfiguration.class);
+
+        assertThat(configuration.list, hasItems("one"));
+    }
+
+    /**
+     * Simple property holder for this test.
+     *
+     * @author Henning Treu - initial contribution
+     *
+     */
+    static class SimpleTestConfiguration {
+        String string;
+        Integer integer;
+        Boolean bool;
+        List<String> list;
+    }
+
+}
```

Comment:
There is already a test suite for ConfigMapper. See https://github.com/eclipse/smarthome/blob/ef748dc460f556e7dd373235282da6ab472191ba/bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/ConfigurationTest.java

Reply:
Thanks, I will move my test case there.